File: javascript-looping-triangle/looping-triangle.js
'use strict'

class LoopingTriangle {
  constructor(triangleLength) {
    this.triangleLength = triangleLength
    this.triangle = ['#'];
    this.counter = 0;
  }

  drawLoopingTriangle() {
    while(this.triangleLength > this.counter) {
      this.triangle.unshift(' ');
      this.counter += 1;
    }
    this.counter = 0;
    while(this.triangleLength > this.counter) {
      console.log(this.triangle.join(" "));
      this.triangle.shift(' ');
      this.triangle.push('#');
      this.counter += 1;
    }
  }
}

module.exports = LoopingTriangle;

File: ruby-looping-triangle/looping_triangle.rb
class LoopingTriangle

  def triangle(triangle_length)
    marker = 0
    hashtag = ["#"]

    until marker == triangle_length
      puts hashtag.join
      hashtag.push("#")
      marker += 1
    end
  end

  def reverse_triangle(triangle_length)
    marker = 0
    hashtag = ["#"]

    triangle_length.times do
      hashtag.unshift(' ')
    end

    until marker == triangle_length
      puts hashtag.join
      hashtag.shift
      hashtag.push('#')
      marker += 1
    end
  end
end

tri = LoopingTriangle.new
tri.triangle(30)
tri.reverse_triangle(30)
